creds/tokens/credentials.go

```diff
@@ -77,37 +62,48 @@ func (c *cred) GetBox(ctx context.Context, address *object.Address) (*accessbox.
 		return nil, err
 	}
 
-	if err = c.cache.Put(address, cachedBox); err != nil {
+	if err = c.cache.Put(addr, cachedBox); err != nil {
 		return nil, err
 	}
 
 	return cachedBox, nil
 }
 
-func (c *cred) getAccessBox(ctx context.Context, address *object.Address) (*accessbox.AccessBox, error) {
-	var (
-		box accessbox.AccessBox
-		buf = c.acquireBuffer()
-	)
-	defer c.releaseBuffer(buf)
+func (c *cred) getAccessBox(ctx context.Context, addr *address.Address) (*accessbox.AccessBox, error) {
+	// init payload reader
+	res, err := c.pool.GetObject(ctx, *addr)
+	if err != nil {
+		return nil, fmt.Errorf("client pool failure: %w", err)
+	}
 
-	ops := new(client.GetObjectParams).WithAddress(address).WithPayloadWriter(buf)
+	defer res.Payload.Close()
 
-	_, err := c.pool.GetObject(
-		ctx,
-		ops,
-	)
-	if err != nil {
-		return nil, err
+	// read payload
+	var data []byte
+
+	if sz := res.Header.PayloadSize(); sz > 0 {
+		if sz <= 4096 { // configure?
```

Comment:
It looks like minor optimization. Suggest to `io.ReadAll()` in all cases. 

By the way, why not `data = make([]byte, sz)` in all cases as well?

Reply:
We limit buffer size when we copy payload to some `io.Writer`.